tensorboard/plugins/histogram/histograms_plugin.py

```diff
@@ -51,16 +68,62 @@ def is_active(self):
     return bool(self._multiplexer) and any(self.index_impl().values())
 
   def index_impl(self):
-    return {
-        run_name: run_data[event_accumulator.HISTOGRAMS]
-        for (run_name, run_data) in self._multiplexer.Runs().items()
-        if event_accumulator.HISTOGRAMS in run_data
-    }
+    """Return {runName: {tagName: {displayName: ..., description: ...}}}."""
+    runs = self._multiplexer.Runs()
+    result = {run: {} for run in runs}
+
+    # Old-style runs
+    for (run, run_data) in six.iteritems(runs):
+      for tag in run_data.get(event_accumulator.HISTOGRAMS, []):
+        result[run][tag] = {'displayName': tag, 'description': ''}
+
+    # New-style runs
+    mapping = self._multiplexer.PluginRunToTagToContent(metadata.PLUGIN_NAME)
+    for (run, tag_to_content) in six.iteritems(mapping):
+      for (tag, content) in six.iteritems(tag_to_content):
+        content = metadata.parse_summary_metadata(content)
+        result[run][tag] = {'displayName': content.display_name,
+                            'description': content.description}
+
+    return result
 
   def histograms_impl(self, tag, run):
-    """Result of the form `(body, mime_type)`."""
-    values = self._multiplexer.Histograms(run, tag)
-    return (values, 'application/json')
+    """Result of the form `(body, mime_type)`, or `ValueError`."""
+    #
+    # Reconcile results from old-style and new-style runs (preferring
+    # new-style runs if there's a conflict).
+    tensor_events = None
+    try:
+      tensor_events = self._multiplexer.Tensors(run, tag)
+    except KeyError:
+      try:
+        tensor_events = self._multiplexer.Histograms(run, tag)
```

Comment:
I did it this way so that the `try` does not enclose the call to `_convert_old_style_histogram_event`, which could potentially shadow errors. However, I can get the best of both worlds with `old_style_events = self._multiplexer.Histograms(run, tag)` and then `else: tensor_events = [… for ev in old_style_events]`. Will change.